bd-session-replay/src/lib.rs

```diff
@@ -0,0 +1,152 @@
+// shared-core - bitdrift's common client/server libraries
+// Copyright Bitdrift, Inc. All rights reserved.
+//
+// Use of this source code is governed by a source available license that can be found in the
+// LICENSE file or at:
+// https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt
+
+#[cfg(test)]
+#[path = "./recorder_test.rs"]
+mod recorder_test;
+
+use bd_runtime::runtime::{session_replay, BoolWatch, ConfigLoader, DurationWatch};
+use bd_shutdown::{ComponentShutdown, ComponentShutdownTrigger};
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::sync::mpsc::{channel, Receiver, Sender};
+use tokio::time::{Instant, Interval};
+
+#[cfg(test)]
+#[ctor::ctor]
+fn test_global_init() {
+  bd_test_helpers::test_global_init();
+}
+
+//
+// Target
+//
+
+pub trait Target {
+  fn capture_wireframes(&self);
+
+  fn take_screenshot(&self);
+}
```

Comment:
How is the platform code going to work here? Is this going to block or be async in some way? Can you add some comments? If it's async we come back to my other question about spamming the channel.